cql/DashboardLibrary.cql

```diff
@@ -162,6 +167,20 @@ define function CytologyInterpretation(conclusions List<System.Concept>):
         when c ~ "ASC-H" then {text: 'ASC-H', rank: 5}
         when c ~ "AGC" then {text: 'AGC', rank: 6}
         when c in "HSIL" or c ~ "SCC" then {text: 'HSIL+', rank: 7}
+        when c ~ "AGC Favor Neoplasia" then {text: 'AGC-FN', rank: 8}
+        when c ~ "Endocervical Cells Favor Neoplasia" then {text: 'AEC-FN', rank: 8}        
+        when c in "AIS" then {text: 'AIS', rank: 9}
```

Comment:
What is rationale behind some Rare Cytologies getting a rank of 8 or 9, and then the rest getting a rank of 1? Should they all have a rank of 1?

Reply:
@mrnosal This is an outstanding issue.

Reply:
This is based on input from the pilot. It will not affect single results from a test, and the rare cytology logic isn't updated to respect rankings for multiple results yet. These ranks can be adjusted once we determine how we intend to include ranking information in the rare cytology logic.

Reply:
First of all, committing this in its current state seems confusing. What is this input from the pilot? Why are the results grouped as they are right now? It seems like we're committing work in progress code without thoroughly documenting why things were changed to be the way they are for the time being.

Secondly, from a logic perspective, what happens now if a patient's second most recent cytology test now has a result of "Endocervical Cells Favor Neoplasia" and "HSIL"? The endocervical cells favor neoplasia would beat out HSIL in the `CytologyInterpretation` function when determining the `riskTableInput`. Since there is no recommendation for a second most recent result of endocervical cells favor neoplasia, this patient would now get no recommendation, instead of a recommendation from the risk table to colpo or treat their HSIL diagnosis.

I do not think we should be adding highly ranked test results to this function without adding comprehensive testing for the scenarios it could affect.

Reply:
The only other option is to define all rare cytologies with rank 1 until there is an agreed solution to ranking with respect to other high ranked cytologies.

Reply:
In addition to changing the rankings, it is confusing to me why the rare cytologies are grouped as they are currently within the `CytologyInterpretation` function. I would suggest we either group all of the rare cytologies together for now, or if we're going to keep them in separate sections, then include more detailed comments about why they are grouped the way that they are.

Reply:
Rare cytologies were grouped according to Dr. Updike's recommendations - there are rare cytologies which were 'more significant' and 'less significant' and this was the first attempt at mapping their rankings accordingly.  There was no additional input regarding the rankings of the less significant rare cytologies, so that's why they were left at 1. 

I've now reset all the rare cytologies back to rank 1 until we come up with a solution for prioritizing the rare cytologies by rank.

Reply:
I see that all cytology results are grouped together and ranked as `1` in https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/134/commits/585e74f0d8198ab53e41769d8f2acbf87d059520. This looks good for now - should accomplish the job of showing the display value for a rare cytology result without affecting the logic yet. I think when we go to re-rank these in a follow-up PR, we should be clear about documenting why certain results were ranked higher than others, especially if the order came from expert opinion as opposed to directly from the guidelines.